### PR TITLE
Add eslint-plugin-prettier

### PR DESCRIPTION
**Pull request by Draydrew**
With this ESLint plugin code gets auto formatted to a particular style on saving. I've tried to set it to as close to our current standards as possible but there are a few differences which can't be dealt with such as imports and JSX props sticking to a single line so long as there's room.

Some standard eslint rules can conflict with how prettier works, so I've also extended the recommended config for it and removed those rules from our end.

Prettier also auto formats and ruins some of the bad pattern examples we had so I just removed those. You do have the ability to use // prettier-ignore in an actual project (https://github.com/prettier/prettier#excluding-code-from-formatting).

There's also a few package updates but nothing really of note.

I'd say this qualifies as a breaking change so I've versioned it as 2.0.0.

P.S. The reason `jest/valid-expect` was missing is because it isn't on the current version; the npm release is v19.0.1, the rule is added in v19.0.2.

### CODE REVIEW

File: eslint.js [hunk 1]
```diff
@@ -39,24 +39,10 @@ module.exports = {
...
+    plugins: ['flowtype', 'import', 'promise', 'eslint-comments', 'babel', 'moment-utc', 'jest', 'jsdoc', 'prettier'],
     settings: {
-        'import/ignore': [
-            'node_modules',
-            '.png',
-            '.jpg',
-            '.json',
-            '.html',
-        ],
+        'import/ignore': ['node_modules', '.png', '.jpg', '.json', '.html'],
```
Comment (DaleLJefferson) [inline eslint.js:45]: Does prettier work on json files?

File: eslint.js [hunk 2]
```diff
@@ -495,5 +387,20 @@ module.exports = {
         'jsdoc/require-param-type': 'error',
         'jsdoc/require-returns-description': 'error',
         'jsdoc/require-returns-type': 'error',
+
+        // eslint prettier
+        'prettier/prettier': [
+            'error',
+            {
+                useTabs: false,
+                printWidth: 120,
```
Comment (DaleLJefferson) [inline eslint.js:388]: Does the whole team agree on 120 width?

**Review by DaleLJefferson — APPROVED** [review]

File: eslint.js (hunk 2 above)
Comment (fjmorant) [reply eslint.js:388]: It is fine for me

**Review by fjmorant — APPROVED** [review]

File: eslint.js (hunk 2 above)
Comment (Draydrew) [reply eslint.js:388]: We've been using 120 for a while already too

File: eslint.js (hunk 1 above)
Comment (Draydrew) [reply eslint.js:45]: I'm not sure, would have to test

File: eslint.js (hunk 2 above)
Comment (airondumael) [reply eslint.js:388]: 120 is fine for me.

Comment (peterallanmdavid) [reply eslint.js:388]: 120 👍

**Review by peterallanmdavid — APPROVED** [review]

**Review by airondumael — APPROVED** [review]

File: eslint.js (hunk 1 above)
Comment (Draydrew) [reply eslint.js:45]: Yep it does

```
// Before
{
    "test": {"more tests": "a","more tests2": 2,"another test": 4}
}

// After
{
    "test": {
        "more tests": "a",
        "more tests2": 2,
        "another test": 4
    }
}
```

**Pull request merged**